op-node/sources/l1_client.go

```diff
@@ -76,6 +76,10 @@ func NewL1Client(client client.RPC, log log.Logger, metrics caching.Metrics, con
 // Notice, we cannot cache a block reference by label because labels are not guaranteed to be unique.
 func (s *L1Client) L1BlockRefByLabel(ctx context.Context, label eth.BlockLabel) (eth.L1BlockRef, error) {
 	info, err := s.BSCInfoByLabel(ctx, label)
+	if err != nil && label == eth.Finalized {
```

Comment:
It should check the error content and only fallback when the error is "methodNotFoundError".

Whenever the error is timeout, service unavailable, or anything else, the error should be thrown outside. Otherwise it may get an unexpected finalized block.